server/protocols/jmap-rfc-8621/src/test/scala/org/apache/james/jmap/routes/JMAPApiRoutesTest.scala

```diff
@@ -67,7 +68,11 @@ object JMAPApiRoutesTest {
   private val mailboxManager: MailboxManager = MemoryMailboxManagerProvider.provideMailboxManager(empty_set)
   private val authenticationStrategy: BasicAuthenticationStrategy = new BasicAuthenticationStrategy(usersRepository, mailboxManager)
   private val AUTHENTICATOR: Authenticator = Authenticator.of(new RecordingMetricFactory, authenticationStrategy)
-  private val JMAP_API_ROUTE: JMAPApiRoutes = new JMAPApiRoutes(AUTHENTICATOR, SERIALIZER)
+
+  private val JMAP_METHODS: java.util.Set[Method] = new java.util.HashSet()
+  JMAP_METHODS.add(new CoreEchoMethod)
+
+  private val JMAP_API_ROUTE: JMAPApiRoutes = new JMAPApiRoutes(AUTHENTICATOR, SERIALIZER, JMAP_METHODS)
```

Comment:
please provide a constructor with a scala immutable set

Reply:
The bindings with Guice are still done in Java, which is why the inject in `JMAPApiRoutes` was taking a `java.util.Set`. But maybe it's not necessary? I can try... still confused a bit between java - scala compatibility sometimes